csharp_dotnetcore/14.NLP-With-Dispatch/Startup.cs

```diff
@@ -0,0 +1,255 @@
+        /// <seealso cref="QnAMaker"/>
+        /// <seealso cref="TelemetryClient"/>
+        private static BotServices InitBotServices(BotConfiguration config)
+        {
+            TelemetryClient telemetryClient = null;
+            var qnaServices = new Dictionary<string, QnAMaker>();
+            var luisServices = new Dictionary<string, LuisRecognizer>();
+
+            foreach (var service in config.Services)
+            {
+                switch (service.Type)
+                {
+                    case ServiceTypes.Luis:
+                        {
+                            // Create a Luis Recognizer that is initialized and suitable for passing
+                            // into the IBot-derived class (NlpDispatchBot).
+                            // In this case, we're creating a custom class (wrapping the original
+                            // Luis Recognizer client) that logs the results of Luis Recognizer results
+                            // into Application Insights for future anaysis.
+                            var luis = (LuisService)service;
+                            if (luis == null)
+                            {
+                                throw new InvalidOperationException("The Luis service is not configured correctly in your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.AppId))
+                            {
+                                throw new InvalidOperationException("The Luis Model Application Id ('appId') is required to run this sample.  Please update your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.AuthoringKey))
+                            {
+                                throw new InvalidOperationException("The Luis Authoring Key ('authoringKey') is required to run this sample.  Please update your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.SubscriptionKey))
+                            {
+                                throw new InvalidOperationException("The Subscription Key ('subscriptionKey') is required to run this sample.  Please update your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.Region))
+                            {
+                                throw new InvalidOperationException("The Region ('region') is required to run this sample.  Please update your '.bot' file.");
+                            }
+
+                            var app = new LuisApplication(luis.AppId, luis.AuthoringKey, luis.Region);
+                            var recognizer = new MyAppInsightLuisRecognizer(app);
+                            luisServices.Add(luis.Name, recognizer);
+                            break;
+                        }
+
+                    case ServiceTypes.Dispatch:
```

Comment:
One thing that's missing here - we're in the "Dispatch" type, yet the comments are all about LUIS. I don't think it's obvious to people that dispatch is built atop LUIS.

A code comment describing this relationship is probably worthwhile.